File: dev-portal/typescript/src/types/environment.ts
/**
 * Environment-related types
 */

/**
 * Environment summary (from list endpoint)
 */
export interface Environment {
  id: string;
  uid: string;
  name: string;
  owner?: string;
  createdAt?: string;
  updatedAt?: string;
  isPublic?: boolean;
}

/**
 * Environment details with variables
 */
export interface EnvironmentDetails extends Environment {
  values: EnvironmentVariable[];
}

/**
 * Environment variable
 */
export interface EnvironmentVariable {
  key: string;
  value: string;
  type?: 'default' | 'secret' | 'any';
  enabled?: boolean;
}

/**
 * Create environment request
 */
export interface CreateEnvironmentRequest {
  name: string;
  values?: EnvironmentVariable[];
  workspaceId: string;
}

/**
 * Create environment result
 */
export interface CreateEnvironmentResult {
  success: boolean;
  environment?: Environment;
  error?: string;
}

/**
 * Update environment request
 */
export interface UpdateEnvironmentRequest {
  environmentUid: string;
  name: string;
  values: EnvironmentVariable[];
}

/**
 * Update environment result
 */
export interface UpdateEnvironmentResult {
  success: boolean;
  environment?: Environment;
  error?: string;
}

/**
 * Patch environment operation
 */
export interface PatchEnvironmentOperation {
  op: 'add' | 'replace' | 'remove';
  path: string;
  value?: unknown;
}

/**
 * Environment mapping (source to target)
 */
export interface EnvironmentMapping {
  sourceUid: string;
  targetUid: string;
  name: string;
}

/**
 * Mock environment update result
 */
export interface MockEnvUpdateResult {
  success: boolean;
  environment?: Environment;
  action?: 'created' | 'updated';
  error?: string;
}

File: dev-portal/python/postman_sdk/types/environment.py
"""
Environment-related types
"""

from typing import Literal
from pydantic import BaseModel, Field


VariableType = Literal["default", "secret", "any"]


class Environment(BaseModel):
    """Environment summary (from list endpoint)"""
    id: str
    uid: str
    name: str
    owner: str | None = None
    created_at: str | None = Field(None, alias="createdAt")
    updated_at: str | None = Field(None, alias="updatedAt")
    is_public: bool | None = Field(None, alias="isPublic")

    class Config:
        populate_by_name = True


class EnvironmentVariable(BaseModel):
    """Environment variable"""
    key: str
    value: str
    type: VariableType = "default"
    enabled: bool = True


class EnvironmentDetails(Environment):
    """Environment details with variables"""
    values: list[EnvironmentVariable] = Field(default_factory=list)


class CreateEnvironmentResult(BaseModel):
    """Create environment result"""
    success: bool
    environment: Environment | None = None
    error: str | None = None


class UpdateEnvironmentResult(BaseModel):
    """Update environment result"""
    success: bool
    environment: Environment | None = None
    error: str | None = None


class EnvironmentMapping(BaseModel):
    """Environment mapping (source to target)"""
    source_uid: str
    target_uid: str
    name: str


class MockEnvUpdateResult(BaseModel):
    """Mock environment update result"""
    success: bool
    environment: Environment | None = None
    action: Literal["created", "updated"] | None = None
    error: str | None = None
